### DISCUSSION TITLE
Ports are not available: exposing port TCP 127.0.0.1:443 -> 0.0.0.0:0: failed

### DISCUSSION THREAD

**Discussion by mattbal** [ddev/ddev#5896] [Q&A]
Hey guys, I'm trying to create a local dev environment for Drupal 10, but I keep getting an error.

I run the following commands:
```
mkdir my-drupal10-site
cd my-drupal10-site
ddev config --project-type=drupal10 --docroot=web --create-docroot
ddev start
```

But upon running `ddev start`, I get the following error:

`Error response from daemon: Ports are not available: exposing port TCP 127.0.0.1:443 -> 0.0.0.0:0: failed to connect to /var/run/com.docker.vmnetd.sock: is vmnetd running?: dial unix /var/run/com.docker.vmnetd.sock: connect: connection refused`

I'm using Docker Desktop and macOS. Any ideas on how to fix this error?

**Comment by rfay** [comment]
Hi @mattbal - This is a common problem in recent versions of Docker Desktop. It's even mentioned in the [docs](https://ddev.readthedocs.io/en/latest/users/install/docker-installation/#macos). You have to explicitly allow Docker Desktop to access them. 

> If you get messages like Ports are not available... exposing port failed... is vmnetd running? it means you need to check the “Allow privileged port mapping (requires password)” checkbox in the “Advanced” section of the Docker Desktop configuration. You may have to stop and restart Docker Desktop.

Let me know if that solves the problem for you.

**Reply by mattbal** [reply]
Hi,
Thanks for getting back to me. I checked my settings and I have that box checked in my Docker Desktop, but I was still getting the error. I tried checking the box above it as well, "Allow the default Docker socket to be used (requires password)" and that fixed the error.
Thanks!
